topography/topography.F90

```diff
@@ -198,14 +205,20 @@ end subroutine topography_init
 !   <ERROR MSG="shape(zmean) is not equal to (/size(blon)-1,size(blat)-1/))" STATUS="FATAL">
 !     Check the input grid size and output field size.
 !   </ERROR>
-#ifndef use_mpp_io
- function get_topog_mean_1d (blon, blat, zmean)
+
+ function get_topog_mean_1d (blon, blat, zmean, use_mpp_io_arg)
 
    real, intent(in),  dimension(:)   :: blon, blat
    real, intent(out), dimension(:,:) :: zmean
    logical :: get_topog_mean_1d
+   logical,optional :: use_mpp_io_arg
 
 !-----------------------------------------------------------------------
+   if (use_mpp_io .OR. present(use_mpp_io_arg) .AND. use_mpp_io_arg) then
```

Comment:
I think you are missing a set of parentheses
```fortran
if (use_mpp_io .OR. (present(use_mpp_io_arg) .AND. use_mpp_io_arg)) then
```
I think stylistically matching how other modules work would be:
```fortran
   logical,optional :: use_mpp_io_arg
   logical :: mpp_io_use
   if (present(use_mpp_io_arg) then
     mpp_io_use = use_mpp_io_arg
   else
     mpp_io_use = use_mpp_io
  endif
!-----------------------------------------------------------------------
￼   if (mpp_io_use) then
```

Reply:
why is `use_mpp_io_arg` even needed? 

`get_topog_mean` is called outside of this subroutine, so `use_mpp_io_arg` is never going to be set anyway (unless we add it) ... and because the if statements are before the `topography_init` call (where use_mpp_io is read from the namelist) fms_io will always be used (at least in the first call)

I think this subroutines needs to be something like: 
(you can probably clean it up farther)
```F90
   if (.not. module_is_initialized) call topography_init()

   if ( any(shape(zmean(:,:)) /= (/size(blon(:))-1,size(blat(:))-1/)) ) &
        call error_mesg('get_topog_mean_1d','shape(zmean) is not&
            & equal to (/size(blon)-1,size(blat)-1/))', FATAL)

 if (use_mpp_io) then
   if ( open_topog_file_mpp_io(topog_file) ) then
       call interp_topog_1d_mpp_io ( blon, blat, zmean )
       get_topog_mean_1d  = .true.
   else
       get_topog_mean_1d = .false.
   endif
else !use_mpp_io
   if ( open_topog_file() ) then
       call interp_topog_1d ( blon, blat, zmean )
       get_topog_mean_1d = .true.
   else
       get_topog_mean_1d = .false.
   endif
endif !use_mpp_io
```
Then there is no need for `get_topog_mean_1d_mpp_io` ... 
` open_topog_file` is where the actual io calls happen.

You can probably get rid of `interp_topog_1d_mpp_io` as well and just wrapped the `input_data`/`input_data_mpp_io` calls (which is actual io) in if statements as well.

Reply:
I mainly added that argument because I thought we were supposed to be adding an argument to public functions to use mpp_io so the version can be selected when called.

My thought process with the mpp_io versions was to just have the fms2_io versions call it's mpp_io version and then any other calls from there would all mpp_io, but I see now that overcomplicated things a bit when only certain calls really need the mpp_io code.